python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -17,6 +17,30 @@
 from cudf.utils.string import format_bytes
 
 
+def get_buffer_owner(data) -> Optional[BufferOwner]:
+    """Get the owner of `data`, if one exists
+
+    Search through the stack of data owners in order to find an
+    owner BufferOwner (incl. subclasses).
+
+    Parameters
+    ----------
+    data
+        The data object
```

Comment:
Would be good to clarify what types are acceptable for `data` (ideally with typing, otherwise here)

Reply:
Added `data: Any` and some more doc